src/throttle/amoc_throttle_config.erl

```diff
@@ -0,0 +1,157 @@
+    #{rate => 0, interval => ?DEFAULT_INTERVAL};
+verify_config(#{interarrival := 0} = Config)
+  when 1 =:= map_size(Config) ->
+    #{rate => infinity, interval => ?DEFAULT_INTERVAL};
+verify_config(#{interarrival := Interarrival} = Config)
+  when 1 =:= map_size(Config), ?POS_INT(Interarrival) ->
+    #{rate => ?DEFAULT_INTERVAL div Interarrival, interval => ?DEFAULT_INTERVAL};
+verify_config(#{rate := Rate, interval := Interval} = Config)
+  when 2 =:= map_size(Config), ?TIMEOUT(Rate), ?NON_NEG_INT(Interval) ->
+    Config;
+verify_config(#{rate := Rate} = Config)
+  when 1 =:= map_size(Config), ?TIMEOUT(Rate) ->
+    Config#{interval => ?DEFAULT_INTERVAL};
+verify_config(_Config) ->
+    {error, invalid_throttle}.
+
+-spec verify_gradual_config(amoc_throttle:gradual_plan()) -> gradual_plan() | {error, any()}.
+verify_gradual_config(Config) ->
+    try do_verify_gradual_config(Config) of
+        Change -> Change
+    catch error:Reason:Stacktrace ->
+              ?LOG_WARNING(#{what => bad_gradual_config,
+                            reason => Reason, stacktrace => Stacktrace}),
+              {error, Reason}
+    end.
+
+-spec pool_config(amoc_throttle:rate(), amoc_throttle:interval()) -> pool_config().
+pool_config(infinity, _) ->
+    Config = #{max_n => infinity, delay => 0, status => active, pid => undefined},
+    maps:from_keys(lists:seq(1, no_of_processes()), Config);
+pool_config(0, _) ->
+    Config = #{max_n => 0, delay => infinity, status => active, pid => undefined},
+    maps:from_keys(lists:seq(1, no_of_processes()), Config);
+pool_config(Rate, 0) ->
+    Config = #{max_n => Rate, delay => 0, status => inactive, pid => undefined},
+    PoolConfig = #{1 := First} = maps:from_keys(lists:seq(1, no_of_processes()), Config),
+    PoolConfig#{1 := First#{status => active}};
+pool_config(Rate, Interval) ->
+    NoOfProcesses = no_of_processes(),
+    RatePerMinutePerProcess = (60000 * Rate div Interval) div NoOfProcesses,
+    DelayPerProcess = (NoOfProcesses * Interval) div Rate,
+    Rem = ((60000 * Rate div Interval) rem NoOfProcesses)
+            + ((NoOfProcesses * Interval) rem Rate),
+    calculate_availability(RatePerMinutePerProcess, DelayPerProcess, NoOfProcesses, Rem).
+
+-spec process_pool_config(pid(), pool_config()) -> pool_config().
+process_pool_config(PoolSup, PoolConfig) ->
+    Workers = amoc_throttle_pool:get_workers(PoolSup),
+    Fun1 = fun(N, Config) -> Config#{pid => maps:get(N, Workers)} end,
+    maps:map(Fun1, PoolConfig).
+
+-spec calculate_availability(integer(), integer(), pos_integer(), integer()) -> pool_config().
+calculate_availability(RatePerMinutePerProcess, DelayPerProcess, NoOfProcesses, Rem) ->
+    Fun = fun(N, {Acc, R}) ->
+                  case {RatePerMinutePerProcess < NoOfProcesses, R} of
+                      {true, 0} ->
+                          Config = #{max_n => RatePerMinutePerProcess,
+                                     delay => DelayPerProcess + 1,
```

Comment:
why `Rem` is added to `delay` but not `max_n`

Reply:
I really don't get this function

Reply:
Let me try to add some comments with explanations.

Reply:
Ok, I've added a property-based test to check that the aggregated delays given to all processes don't differ from the actually requested rate by more than 1%, and I cannot get it to pass 😱 

Please have a look at `throttle_SUITE:pool_config_property_tests/1` @DenysGonchar